### PR TITLE
Added procedure to Evaluate performance, needs more work

### PR DESCRIPTION
**Pull request by RohitX0X**
As we keep developing it is also important to evaluate the performance, This motivates in improving model performance and experimentation with the current pipeline

I have created a basic evaluation folder, which follows idea of regression testing. Would improve more once i get free...

Procedure:
It has 5 Prompts which is tested using devika , with a llm model as user input 

The goal idea is to provide information on metrics like : how many tasks have passed, average Openai API hit count, avg token count, avg time of execution  

Please fork/PR for contribution for evaluations

### CODE REVIEW

File: evaluation/evaluation.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        self.tokenusage = 0
+        self.averagetokenperstep = 0
+        self.stepcount=0 
+        self.llmcalls=0
+
+if __name__ == '__main__':
+    instructions_data = load_tasks()
+    instructions = instructions_data['instructions']
+    models = get_model_list()
+    model = models['models'][5][1]
```
Comment (RohitX0X) [inline evaluation/evaluation.py:70]: replace with your model id

**Comment by RohitX0X** [comment]
Replace model id , appropriately, 
in evaluation folder:
`python evaluate.py`

Also beware as testing process will use up lot of your tokens 

 Improvements required:
1) choose model id by choosing maybe a small pop up box
2) Complete required evaluation metrics
3) replace Async or sync calls wherever required

**Comment by mufeedvh** [comment]
This looks good to me, will add this to the testing suite once that's planned properly. Closing this for now, but will re-open or link when the testing suite project is initiated. Thanks! ❤️ 🙌

**Comment by RohitX0X** [comment]
Thanks for the feedback man, let me know when its on

**Pull request closed without merging**